### PR TITLE
Adding Doc for Upgrades Addon

### PR DESCRIPTION
**Pull request by Guillaume-Lebegue**
This is the Documentation for the soon to be released Upgrades Addon.

If someone could look at it and give me feedback.

### CODE REVIEW

File: docs/addons/Upgrades/index.md [hunk 1]
```diff
@@ -0,0 +1,298 @@
...
+* command-icon
+
+!!! tip
+    All `upgrade`, `island-min-level` and `vault-cost` fields are mathematical expression. So:
+
+    * +,-,*,/,^,(,) can be used
+    * sqrt(), sin(), cos(), tan() can be used
+    * `[level]` Is replaced by the actual level for this upgrade
+    * `[islandLevel]` Is replaced by the island level from level addon **(Can be 0)**
+    * `[numberPlayer]` Is replaced by the number of players in the team
```
Comment (Poslovitch) [inline docs/addons/Upgrades/index.md:49]:
```suggestion
    * `[level]` is replaced by the actual level for this upgrade
    * `[islandLevel]` is replaced by the island level from level addon **(Can be 0)**
    * `[numberPlayer]` is replaced by the number of players in the team

```

File: docs/addons/Upgrades/index.md [hunk 2]
```diff
@@ -0,0 +1,298 @@
...
+  island-min-level: "2"
+  vault-cost: "[level]*100"
+  permission-level: 1
+```
+
+* `max-level` Is the maximum level of this tier.
+* `upgrade` Is how much is given at each level.
+* `island-min-level` Is the minimum island level needed to buy this upgrade. It is given by [Level Addon](/addons/Level)
+* `vault-cost` Is the cost to buy this upgrade **(>= 0)**
+* `permission-level` Is the level of permission needed to buy this upgrade (CF: Permission)
```
Comment (Poslovitch) [inline docs/addons/Upgrades/index.md:69]:
```suggestion
* `max-level` is the maximum level of this tier.
* `upgrade` is how much is given at each level.
* `island-min-level` is the minimum island level needed to buy this upgrade. It is given by [Level Addon](/addons/Level)
* `vault-cost` is the cost to buy this upgrade **(>= 0)**
* `permission-level` is the level of permission needed to buy this upgrade (cf. Permission)

```

File: docs/addons/Upgrades/index.md [hunk 3]
```diff
@@ -0,0 +1,298 @@
...
+      max-level: 2
+      island-min-level: "2"
+      vault-cost: "[level]*200"
+      console: true
+      command: 
+        - "say [player] has upgrade his lambda to level [level]"
+        - "say [player] has reached the max level"
+```
+
+### GameMode
```
Comment (Poslovitch) [inline docs/addons/Upgrades/index.md:190]:
```suggestion
### Gamemode

```

File: docs/addons/Upgrades/index.md [hunk 4]
```diff
@@ -0,0 +1,298 @@
...
+    tier2:
+      max-level: 5
+      upgrade: "1"
+      island-min-level: "4"
+      vault-cost: "([level]-2)*[numberPlayer]*700"
+      permission-level: 1
+```
+
+!!! tip "Blocks"
+    A list of blocks can be found [here](https://hub.spigotmc.org/javadocs/bukkit/org/bukkit/Material.html) (1.16.1)
```
Comment (Poslovitch) [inline docs/addons/Upgrades/index.md:119]:
```suggestion
    A list of blocks can be found [here](https://hub.spigotmc.org/javadocs/bukkit/org/bukkit/Material.html)

```

Not needed here

**Comment by Guillaume-Lebegue** [comment]
That should be good

**Pull request merged**